pkg/operator/ceph/cluster/mon/drain.go

```diff
@@ -108,10 +108,18 @@ func (c *Cluster) allowMonDrain(request types.NamespacedName) error {
 		return nil
 	}
 	logger.Info("allow voluntary mon drain after failover")
-	// change MaxUnavailable mon PDB to 1
-	_, err := c.createOrUpdateMonPDB(1)
+	_, err := c.createOrUpdateMonPDB(c.getMaxUnavailableMonPodCount())
 	if err != nil {
 		return errors.Wrapf(err, "failed to update MaxUnavailable for mon PDB %q", request.Name)
 	}
 	return nil
 }
+
+func (c *Cluster) getMaxUnavailableMonPodCount() int32 {
+	if c.spec.Mon.Count >= 5 {
+		logger.Debug("setting the mon pdb max unavailable count to 2 in case there are 5 or more mons")
+		return 2
```

Comment:
If there are 7 mons, this could return 3, or 9 mons could be 4. But I don't believe those mon counts are ever used, so looks good to keep this simple.